distributed/protocol/keras.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import print_function, division, absolute_import
+
+from .serialize import register_serialization, serialize, deserialize
+
+from ..utils import log_errors, ensure_bytes
+
+
+def serialize_keras_model(model):
+    header = model._updated_config()
+    weights = model.get_weights()
+    headers, frames = list(zip(*map(serialize, weights)))
+    header['headers'] = headers
+    header['lengths'] = [len(L) for L in frames]
+    frames = [frame for L in frames for frame in L]
+    return header, frames
```

Comment:
OK, the failure here is my fault.  Dask is already using the term `'lengths'` and so there is a conflict.  Naming this something else should work.  However I've also identified another internal issue.  Resolving that now.